sdk/storage/Azure.Storage.Blobs/src/PartitionedDownloader.cs

```diff
@@ -387,8 +388,11 @@ private static long ParseRangeTotalLength(string range)
             CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
```

Comment:
I wonder if we should use this opportunity to normalize on throwing a TaskCanceledException rather than OperationCanceledException.

In the PR for RetriableStream, I'm using the CancellationHelper to throw a TaskCanceledException - https://github.com/Azure/azure-sdk-for-net/pull/26822/files#diff-062a7f5473fad3dfdf10f00a2efe8b18a2a6b7f5b79c4c6e220e19ef942e489bR115

Reply:
I'm unsure of the conventions here. If we want to change it we can, I'm just not sure what the correct choice is.

Reply:
Well, Azure.Core normalizes to throwing TaskCanceledExceptions, so it would probably be better to follow that pattern here so users don't get TaskCanceledExceptions sometimes, and OperationCanceledException other times. Changing from OperationCanceledException to TaskCanceledException is allowed since TaskCanceledException inherits from OperationCanceledException.